cmake/modules/DispatchCompilerWarnings.cmake

```diff
@@ -0,0 +1,85 @@
+
+macro(dispatch_common_warnings)
```

Comment:
these are the flags we have enabled in the darwin build internally (in Xcode), so shared code should be built with these enabled to avoid breaking the darwin build (and I want the darwin CMake build to definitely have these so that it matches the Xcode build as closely as possible).
We maintain these flags internally by hand by adding or removing new ones as clang adds support for them (based on -Weverything).

If we need to opt out non-clang platforms like Windows that we don't build & support on a continual basis right now, that is a much lesser concern IMO